subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/fixture/TestNGCoverage.groovy

```diff
@@ -16,20 +16,53 @@
 
 package org.gradle.testing.fixture
 
-import com.google.common.collect.ObjectArrays
+import org.gradle.api.JavaVersion
 import org.gradle.integtests.fixtures.RepoScriptBlockUtil
-import org.gradle.internal.jvm.Jvm
+import org.gradle.util.internal.VersionNumber
 
 class TestNGCoverage {
     final static String NEWEST = '7.5'
-    final static String INITIAL_BROKEN_ICLASS_LISTENER = '6.9.10' // introduces initial, buggy IClassListener
-    final static String FIXED_ICLASS_LISTENER = '6.9.13.3' // introduces fixed IClassListener
-    final static String[] STANDARD_COVERAGE = ['5.14.10', '6.2', '6.8.7', '6.9.13.6', NEWEST]
-    final static String[] STANDARD_COVERAGE_WITH_INITIAL_ICLASS_LISTENER =  ObjectArrays.concat(INITIAL_BROKEN_ICLASS_LISTENER, STANDARD_COVERAGE)
-    final static String LEGACY =  '5.12.1' // lacks TestNG#setConfigFailurePolicy method
-    final static String[] STANDARD_COVERAGE_WITH_LEGACY =  ObjectArrays.concat(LEGACY, STANDARD_COVERAGE_WITH_INITIAL_ICLASS_LISTENER)
-    final static String[] PRESERVE_ORDER = Jvm.current().javaVersion.java7Compatible ? ['5.14.6', '6.1.1', '6.9.4', NEWEST] : ['5.14.6', '6.1.1'] // skipped NEWEST (6.8.7) because of cbeust/testng#639
-    final static String[] GROUP_BY_INSTANCES = ['6.1', '6.8.7', NEWEST]
+
+    private static final String FIXED_ILLEGAL_ACCESS = '5.14.6' // Oldest version to support JDK 16+ without explicit --add-opens
+    private static final String FIXED_ICLASS_LISTENER = '6.9.13.3' // Introduces fixed IClassListener
+
+    private static final Set<String> VERSIONS = [
+        '5.12.1', // Newest version without TestNG#setConfigFailurePolicy method (Added in 5.13)
+        FIXED_ILLEGAL_ACCESS,
+        '6.1.1',  // Fixes NPE when using preserve-order and factories.
+        '6.8.21', // Newest version with cbeust/testng#639 bug
+        '6.9.4',  // Fixes cbeust/testng#639 for preserve-order
+        '6.9.10', // Introduces initial, buggy IClassListener
+        FIXED_ICLASS_LISTENER,
+        NEWEST
+    ]
+
+    static final Set<String> SUPPORTED_BY_JDK
+    static final Set<String> SUPPORTS_PRESERVE_ORDER
+    static final Set<String> SUPPORTS_GROUP_BY_INSTANCES
+    static final Set<String> SUPPORTS_ICLASS_LISTENER
+
+    static {
```

Comment:
Why the static block here vs. declare + assign on the same line for these fields?

Reply:
Was following the direction of `GroovyCoverage`, but have removed the `static` block since it is not necessary